src/rescue_password_menu.c

```diff
@@ -635,8 +635,8 @@ u32 ConvertMenutoRescuePasswordState(u32 unused)
 u32 sub_8039068(u32 mailMode, u8 *passwordBuffer, struct unkStruct_203B480 *param_3)
 {
   if ( (!sub_803D204(passwordBuffer, param_3)) || (WONDER_MAIL_TYPE_OKD < param_3->mailType) ||
-       (param_3->dungeon.floor >= GetDungeonFloorCount(param_3->dungeon.id)) ||
-       (param_3->clientSpecies == MONSTER_NONE) || (MONSTER_MAX < param_3->clientSpecies) ||
+       (param_3->unk4.dungeon.floor >= GetDungeonFloorCount(param_3->unk4.dungeon.id)) ||
+       (param_3->clientSpecies == MONSTER_NONE) || (MONSTER_RAYQUAZA_CUTSCENE < param_3->clientSpecies) ||
```

Comment:
```suggestion
       (param_3->clientSpecies == MONSTER_NONE) || (MONSTER_MAX< param_3->clientSpecies) ||
```

I derped on a rebase. Should be this.